lib/dispatcher/proxy-agent.js

```diff
@@ -40,7 +40,9 @@ class ProxyClient extends DispatcherBase {
 
     super()
 
-    this.#client = new Client(origin, opts)
+    // Store the connector for direct socket connections
+    this.#connector = opts?.connect || buildConnector(opts)
+    this.#client = new Pool(origin, opts)
```

Comment:
the new Http1ProxyWrapper does this, although looking at the diff it's kind of clumbsy how it does it -- but the default factory it uses will make a Pool unless opts.connections === 1, which is pulled from the regular Agent factory.

I did switch to using a user supplied factory in order to get MockAgent tests passing, though.

Reply:
Then this PR is not needed